File: articles/future-directions-in-decision-science/python/run_all_future_decision_science_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Future Directions in Decision Science."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "future_maturity_model.py",
    "ai_reliance_model.py",
    "adaptive_review_model.py",
    "future_pathway_comparison.py",
    "decision_record_exporter.py",
    "future_decision_science_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python future decision-science workflows completed.")


if __name__ == "__main__":
    main()

File: articles/future-directions-in-decision-science/r/run_all_future_decision_science_workflows.R
# run_all_future_decision_science_workflows.R
# Run all R workflows for Future Directions in Decision Science.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "future_directions_decision_science_workflow.R",
  "future_pathway_profiles.R",
  "failure_risk_review.R",
  "adaptive_capacity_review.R",
  "future_decision_science_summary.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R future decision-science workflows completed.")
